custom-metrics-stackdriver-adapter/adapter-staging.yaml

```diff
@@ -138,3 +138,30 @@ spec:
     name: custom-metrics-stackdriver-adapter
     namespace: custom-metrics
   version: v1beta1
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: external-metrics-reader
+rules:
+- apiGroups:
+  - "external.metrics.k8s.io"
+  resources:
+  - "*"
```

Comment:
Does it have to read everything? Is it possible to have external metrics on e.g. custom resources? Also, this will enable reading secrets, is it necessary?

Reply:
Can we go forward with this for now since this is the staging version? I see the alpha adapter version does the same thing and the staging version should only be used for testing at the moment.

Reply:
Well, ok, but please leave a TODO here to have at least a slight chance of fixing this in the future.

Reply:
It has to read all types of resources. In current version of proposal: https://github.com/kubernetes/community/pull/1802, the resource corresponds to metric name, so HPA needs access to all of them. In earlier version of the proposal, resource name was "externalmetrics", but I assume that the implementation of new proposal for HPA is already merged and this won't change. @MaciekPytel to confirm.